src/NServiceBus.Core/Recoverability/Discard.cs

```diff
@@ -0,0 +1,18 @@
+namespace NServiceBus
+{
+    /// <summary>
+    /// Indicates recoverability is required to discard/ignore the current message.
+    /// </summary>
+    public sealed class Discard : RecoverabilityAction
+    {
+        internal Discard(string reason)
+        {
+            Reason = reason;
```

Comment:
Should we have a guard against null and empty?

Reply:
Done